prow/plugins/bugzilla/bugzilla.go

```diff
@@ -819,6 +850,53 @@ func handleMerge(e event, gc githubClient, bc bugzilla.Client, options plugins.B
 	return comment(fmt.Sprintf("%s %s\n%s", mergedMessage("Some"), unmergedMessage, outcomeMessage("")))
 }
 
+func handleCherrypick(e event, gc githubClient, bc bugzilla.Client, options plugins.BugzillaBranchOptions, log *logrus.Entry) error {
+	comment := e.comment(gc)
+	// get the info for the PR being cherrypicked from
+	pr, err := gc.GetPullRequest(e.org, e.repo, e.cherrypickFromPRNum)
+	if err != nil {
+		log.WithError(err).Warn("Unexpected error getting title of pull request being cherrypicked from.")
+		return comment(formatError(fmt.Sprintf("checking the state of a cherrypicked pull request at https://github.com/%s/%s/pull/%d", e.org, e.repo, e.cherrypickFromPRNum), bc.Endpoint(), e.bugId, err))
+	}
+	// Attempt to identify bug from PR title
+	mat := titleMatch.FindStringSubmatch(pr.Title)
+	if mat == nil {
+		log.WithError(err).Debugf("Parent PR %d doesn't have associated bug; not creating cherrypicked bug", pr.Number)
+		return comment(formatError(fmt.Sprintf("Parent PR %d does not have an associated bug; skipping cherrypick bug creation", pr.Number), bc.Endpoint(), e.bugId, err))
+	}
+	bugID, err := strconv.Atoi(mat[1])
+	if err != nil {
+		// should be impossible based on the regex
+		log.WithError(err).Debug("Failed to parse bug ID as int - is the regex correct?")
+		return comment(formatError(fmt.Sprintf("Failed to parse bug ID (%s) as int", mat[1]), bc.Endpoint(), e.bugId, err))
+	}
+	bug, err := getBug(bc, bugID, log, comment)
+	if err != nil {
+		log.WithError(err).Debugf("Failed to get bug %d", bugID)
+		return comment(formatError(fmt.Sprintf("Failed to get bug %d", bugID), bc.Endpoint(), e.bugId, err))
+	}
+	cloneID, err := bc.CloneBug(bug)
```

Comment:
what if the bug already has been cloned? we do not want to clone again nor fail

Reply:
To prevent from cloning again, we could use the bugzilla search API to find bugs with the correct summary:

If the ID is different from the parent ID but the version is the same, then that bug is a clone that failed to have its version updated.

If it has the correct target release, we just failed to link it somehow, and we should link it here.

If it has anything else as the target release, it's probably a different cherry pick that we should ignore.

If we don't match anything above, that means we haven't cloned it yet, so we should just proceed to clone normally.

Reply:
Since this code does not need to be re-entrant I think we can forgo this